### ISSUE TITLE
Excess object literals property checking fails when the literal includes a spread

### ISSUE THREAD

**Issue by sandersn** [Microsoft/TypeScript#13878]
Excess object literal property checking fails when the literal includes a spread.

```ts
let x = { b: 1, extra: 2 }
let xx: { a, b }  = { a: 1, ...x, z: 3 } // error for 'z', no error for 'extra'
```

Actual: No Error

Expected:
Error for 'z', no error for 'extra'

**Comment by sandersn** [comment]
Probably `checkObjectLiteral` needs to set ObjectLiteral and FreshObjectLiteral flags on the result of `getSpreadType` (if the result is an object type &mdash; if it's a type parameter then freshness wouldn't apply anyway).

**Comment by mhegazy** [comment]
We discussed this in https://github.com/Microsoft/TypeScript/issues/12997, and decided it worked as intended. what changed?

**Comment by sandersn** [comment]
I just forgot when we noticed it again on Friday afternoon. On the surface it looks like an oversight.

**Comment by whitecolor** [comment]
```
let x = { b: 1, extra: 2 }
let xx: { a, b }  = { a: 1, ...x, z: 3 } // error for 'z', no error for 'extra'
```

I see currenlty no error for z was that fixed?

And I also don't get why there should not be an error for `extra` if we are trying to assign it to `{a, b}`

**Comment by sandersn** [comment]
@whitecolor See the last section #12997 for discussion.

**Comment by whitecolor** [comment]
So it will be fixed, but in 2.4?

**Comment by mhegazy** [comment]
yes.
